cmd/dra-example-controller/driver.go

```diff
@@ -244,17 +245,15 @@ func (d driver) unsuitableNode(ctx context.Context, pod *corev1.Pod, allcas []*c
 
 	perKindCas := make(map[string][]*controller.ClaimAllocation)
 	for _, ca := range allcas {
-		var kind string
-		switch ca.ClaimParameters.(type) {
-		case *gpucrd.GpuClaimParametersSpec:
-			kind = gpucrd.GpuClaimParametersKind
+		if ca.Claim.Spec.ParametersRef.Kind == gpucrd.GpuClaimParametersKind {
+			perKindCas[gpucrd.GpuClaimParametersKind] = append(perKindCas[gpucrd.GpuClaimParametersKind], ca)
+		} else {
```

Comment:
Idiomatic Go would be to check for the error case first, then continue without indention. if/else hardly ever get used.

In this case, a switch with default would be clearer.